pallet-smart-contract/src/lib.rs

```diff
@@ -605,85 +603,61 @@ impl<T: Config> Module<T> {
     fn _distribute_cultivation_rewards(
         contract: &types::Contract,
-        node: &pallet_tfgrid_types::Node,
-        farm: &pallet_tfgrid_types::Farm,
         pricing_policy: &pallet_tfgrid_types::PricingPolicy<T::AccountId>,
         amount: BalanceOf<T>,
     ) -> DispatchResult {
-        // fetch farmer twin
-        let farmer_twin = pallet_tfgrid::Twins::<T>::get(farm.twin_id);
         // fetch source twin
         let twin = pallet_tfgrid::Twins::<T>::get(contract.twin_id);
 
-        // parse amount
-        let amount = U64F64::from_num(amount.saturated_into::<u128>());
+        // Burn 35%
+        let burned_amount = Perbill::from_percent(35) * amount;
+        <T as Config>::Currency::slash(&twin.account_id, burned_amount);
 
-        let foundation_share = amount * U64F64::from_num(0.3);
-        let mut certified_sales_share = amount * U64F64::from_num(0.6);
-
-        // by default, 10% of the cultivation rewards go to the farmer
-        let mut farmer_share = amount * U64F64::from_num(0.1);
-        // if the farmer is deploying workloads on his own farm, return to him 70% of the total amount
-        if node.farm_id == farm.id {
-            farmer_share = amount * U64F64::from_num(0.7);
-            certified_sales_share = U64F64::from_num(0);
-        }
-
-        // Tranfer to foundation account
-        let foundation_share_balance =
-            BalanceOf::<T>::saturated_from(foundation_share.ceil().to_num::<u128>());
+        // Send 10% to the foundation
+        let foundation_share = Perbill::from_percent(10) * amount;
         debug::info!(
             "Transfering: {:?} from contract twin {:?} to foundation account {:?}",
-            &foundation_share_balance,
+            &foundation_share,
             &twin.account_id,
             &pricing_policy.foundation_account
         );
         <T as Config>::Currency::transfer(
             &twin.account_id,
             &pricing_policy.foundation_account,
-            foundation_share_balance,
+            foundation_share,
             KeepAlive,
         )
         .map_err(|_| DispatchError::Other("Can't make foundation share transfer"))?;
+        
+        // TODO: send 5% to the staking pool account
+        let staking_pool_share = Perbill::from_percent(5) * amount;
+        let staking_pool_account = T::StakingPoolAccount::get();
+        <T as Config>::Currency::transfer(
+            &twin.account_id,
+            &staking_pool_account,
+            staking_pool_share,
+            KeepAlive,
+        )
+        .map_err(|_| DispatchError::Other("Can't make staking pool share transfer"))?;
 
-        // Transfer to farmer account
-        let farmers_share_balance =
-            BalanceOf::<T>::saturated_from(farmer_share.ceil().to_num::<u128>());
+        // Send 50% to the sales channel
+        let sales_share = Perbill::from_percent(50) * amount;
```

Comment:
There is a subtle rounding issue here. The `*` operator actually calls the `mul` method from the `Mul` thread, and while it is not documented, this rounds the outcome based on the regular rounding rules. This is actually documented in the `mul_ceil` and `mul_floor` methods of the `PerThing` trait. (https://docs.substrate.io/rustdocs/latest/sp_runtime/trait.PerThing.html#method.mul_ceil).

Basically, it is possible for previous values to be rounded up, so that the sum of the calculated values will be slightly bigger than the amount. To fix this, the last amount should be calculated as the remainder of "amount - all previous calculated payouts".